lib/json-schema/validator.rb

```diff
@@ -134,6 +134,13 @@ def load_ref_schema(parent_schema, ref)
       schema_uri = absolutize_ref_uri(ref, parent_schema.uri)
       return true if self.class.schema_loaded?(schema_uri)
 
+      begin
+        validator = self.class.validator_for_uri(schema_uri)
+        schema_uri = JSON::Util::URI.file_uri(validator.metaschema)
+      rescue JSON::Schema::SchemaError
```

Comment:
I've corrected that now, by allowing `validator_for_uri` to take a second parameter that allows it to return `nil` (rather than raising an exception).

I chose that implementation (rather than bang methods) so that we don't change the default behaviour of any methods.